cluster-provision/tools/check-image-pull-policies/check-image-pull-policies.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"flag"
+	"fmt"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/serializer"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type options struct {
+	manifestSource string
+	verbose        bool
+	dryRun         bool
+}
+
+func flagOptions() options {
+	o := options{}
+	flag.StringVar(&o.manifestSource, "manifest-source", "", "The directory with manifest files or the manifest file to check. If a directory is given, all files are tried to parse as manifest")
+	flag.BoolVar(&o.dryRun, "dry-run", true, "Whether to exit with a non-zero exit code if the check fails")
+	flag.BoolVar(&o.verbose, "verbose", false, "Whether to output all parsed information or only the information on where the checks failed")
+	flag.Parse()
+	return o
+}
+
+var deserializer = serializer.NewCodecFactory(runtime.NewScheme()).UniversalDeserializer()
+
+func main() {
+	options := flagOptions()
+
+	if options.manifestSource == "" {
+		log.Fatal("No manifest-source given!")
+	}
+
+	fileInfo, err := os.Stat(options.manifestSource)
+	if os.IsNotExist(err) {
+		log.Fatalf("Failed to open %s: %v", options.manifestSource, err)
+	}
+
+	filesWithPullPolicies := map[string]map[string]corev1.PullPolicy{}
+	if fileInfo.IsDir() {
+		err = filepath.Walk(options.manifestSource, func(path string, info os.FileInfo, err error) error {
+			if info.IsDir() {
+				return nil
+			}
+			checkFileForPullPolicies(path, filesWithPullPolicies)
+			return nil
+		})
+		if err != nil {
+			log.Fatalf("Error on walking path %s: %v", options.manifestSource, err)
+		}
+	} else {
+		checkFileForPullPolicies(options.manifestSource, filesWithPullPolicies)
+	}
+
+	fmt.Printf("%d files with pull policies detected\n", len(filesWithPullPolicies))
+	hasOffendingPolicies := false
+	for filePath, pullPolicies := range filesWithPullPolicies {
+		for image, pullPolicy := range pullPolicies {
+			imageParts := strings.Split(image, ":")
+			var imageTag string
+			if len(imageParts) > 0 {
+				imageTag = imageParts[1]
+			}
+			offending := pullPolicy == corev1.PullAlways ||
```

Comment:
Can you please make it a proper condition? It's not a one liner anyway and I think that it's hard to understand it that way. I would even move it into a proper function to easily test it.